include/velocypack/inspection/Inspector.h

```diff
@@ -0,0 +1,333 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Manuel Pöter
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <array>
+#include <functional>
+#include <type_traits>
+#include <variant>
+#include "velocypack/Iterator.h"
+#include "velocypack/inspection/InspectorAccess.h"
+
+namespace arangodb::velocypack::inspection {
+
+template<class Derived>
+struct InspectorBase {
+  Derived& self() { return static_cast<Derived&>(*this); }
+
+  template<class T>
+  [[nodiscard]] Result apply(T& x) {
+    return process(self(), x);
+  }
+
+  template<class T>
+  struct Object;
+
+  template<class T, const char ErrorMsg[], class Func, class... Args>
+  static Result checkInvariant(Func&& func, Args&&... args) {
```

Comment:
can this function be a constexpr? Like 
```suggestion
  static constexpr Result checkInvariant(Func&& func, Args&&... args) {
```

Reply:
maybe is a candidate for
```suggestion
  [[nodiscard]] static constexpr Result checkInvariant(Func&& func, Args&&... args) {
```
?